lib/saluki-components/src/transforms/name_filter/mod.rs

```diff
@@ -0,0 +1,238 @@
+        "cassandra".to_string(),
+        "confluent".to_string(),
+        "hazelcast".to_string(),
+        "hive".to_string(),
+        "ignite".to_string(),
+        "jboss".to_string(),
+        "jvm".to_string(),
+        "kafka".to_string(),
+        "presto".to_string(),
+        "sidekiq".to_string(),
+        "solr".to_string(),
+        "tomcat".to_string(),
+        "runtime".to_string(),
+    ]
+}
+
+impl NameFilterConfiguration {
+    /// Creates a new `NameFilterConfiguration` from the given configuration.
+    pub fn from_configuration(config: &GenericConfiguration) -> Result<Self, GenericError> {
+        Ok(config.as_typed()?)
+    }
+}
+
+#[async_trait]
+impl TransformBuilder for NameFilterConfiguration {
+    fn input_data_type(&self) -> DataType {
+        DataType::Metric
+    }
+
+    fn outputs(&self) -> &[OutputDefinition] {
+        static OUTPUTS: &[OutputDefinition] = &[OutputDefinition::default_output(DataType::Metric)];
+        OUTPUTS
+    }
+
+    async fn build(&self, _: ComponentContext) -> Result<Box<dyn Transform + Send>, GenericError> {
+        Ok(Box::new(NameFilter {
+            metric_prefix: self.metric_prefix.clone(),
+            metric_prefix_blacklist: self.metric_prefix_blacklist.clone(),
+            blocklist: Blocklist::new(&self.metric_blocklist, self.metric_blocklist_match_prefix),
+        }))
+    }
+}
+
+impl MemoryBounds for NameFilterConfiguration {
+    fn specify_bounds(&self, builder: &mut MemoryBoundsBuilder) {
+        // Capture the size of the heap allocation when the component is built.
+        builder.minimum().with_single_value::<NameFilter>();
+    }
+}
+
+#[derive(Debug)]
+struct Blocklist {
+    data: Vec<String>,
+    match_prefix: bool,
+}
+
+impl Blocklist {
+    fn new(data: &[String], match_prefix: bool) -> Self {
+        let mut data = data.to_owned();
+        data.sort();
+
+        if match_prefix && !data.is_empty() {
+            let mut i = 0;
+            for j in 1..data.len() {
+                if !data[j].starts_with(&data[i]) {
+                    i += 1;
+                    data[i] = data[j].clone(); // Move item to next position
+                }
+            }
+            data.truncate(i + 1); // Only keep valid elements
+        }
+
+        Blocklist { data, match_prefix }
+    }
+
+    fn test(&self, name: String) -> bool {
+        if self.data.is_empty() {
+            return false;
+        }
+
+        let i = self.data.binary_search(&name);
+
+        // Check for prefix match when match_prefix is true
+        if self.match_prefix {
+            if let Ok(index) = i {
+                if index > 0 && name.starts_with(&self.data[index - 1]) {
+                    return true;
+                }
+            } else if i.is_err() {
+                let index = i.unwrap_err();
+                if index > 0 && name.starts_with(&self.data[index - 1]) {
+                    return true;
+                }
+            }
```

Comment:
Just use a `match` block to extract `index`, and then you can deduplicate this and keep it closer to the original:

```rust
if self.match_prefix {
    let index = i.map_or_else(|idx| idx, |idx| idx);
    if index > 0 && name.starts_with(&self.data[index - 1]) {
        return true;
    }
}
```

(The `map_or_else` bit is just a condensed form of `match` where we return the contained value for either side of `Result<T, E>` just they're both `usize`.)